hivemind/averaging/averager.py

```diff
@@ -393,20 +417,20 @@ async def _step(
                     asyncio.InvalidStateError,
                     P2PHandlerError,
                 ) as e:
-                    time_elapsed = get_dht_time() - start_time
-                    if not allow_retries or (timeout is not None and timeout < time_elapsed):
-                        logger.exception(f"Averager caught {repr(e)}")
-                        future.set_exception(e)
+                    if not step.allow_retries or get_dht_time() >= step.deadline:
+                        logger.exception(f"{self.__class__.__name__} caught {repr(e)}")
```

Comment:
I'm not sure if any of this is necessary, since we'll have a traceback with the name of the class and repr() of the exception anyway

Reply:
fixd